src/panels/config/entity_registry/ha-config-entity-registry.ts

```diff
@@ -45,9 +45,62 @@ class HaConfigEntityRegistry extends LitElement {
 
   private _filteredEntities = memoize(
     (entities: EntityRegistryEntry[], showDisabled: boolean) =>
-      showDisabled
+      (showDisabled
         ? entities
         : entities.filter((entity) => !Boolean(entity.disabled_by))
+      ).map((entry) => {
+        const state = this.hass!.states[entry.entity_id];
+        return {
+          ...entry,
+          icon: state
+            ? stateIcon(state)
+            : domainIcon(computeDomain(entry.entity_id)),
+          name:
+            computeEntityRegistryName(this.hass!, entry) ||
+            this.hass!.localize("state.default.unavailable"),
+          enabled: !entry.disabled_by,
+          id: entry.entity_id,
+        };
+      })
+  );
+
+  private _columns = memoize(
+    (): DataTabelColumnContainer => {
+      return {
+        icon: {
+          title: "",
+          template: (icon) => html`
```

Comment:
If we can extend this method signature, I would suggest we pass the whole object as the 2nd arg. That way templates can use other info too.

Reply:
👍 That would help for a lot of things!

Reply:
I don't think it is needed in this case, as calculating the icon once might be enough, only thing  you miss is updating the icon if a state changes, but for this use case not as important?

Reply:
Actually. This current method takes no parameters, what is it memorizing on?

Reply:
Nothing, we just need to create it once. It is the same as:
`this._columns = this._columns || this.generateColumns();`

But since I don't use `this` here, I can move it out of the class and make it static.